lib/scripts.js

```diff
@@ -371,6 +371,68 @@ var scripts = {
 
     return execScript.apply(scripts, args);
   },
+
+  cleanJobsInSet: function(queue, set, ts) {
+    var command;
+    var hash;
+    switch(set) {
+      case 'completed':
+      case 'failed':
+        command = 'local JOBS = redis.call("SMEMBERS", KEYS[1])';
+        hash = 'cleanSet';
+        break;
+      case 'wait':
+      case 'active':
+      case 'paused':
+        command = 'local JOBS = redis.call("LRANGE", KEYS[1], 0, -1)';
+        hash = 'cleanList';
+        break;
+      case 'delayed':
+        command = 'local JOBS = redis.call("ZRANGE", KEYS[1], 0, -1)';
+        hash = 'cleanOSet';
+        break;
+    }
+
+    var script = [
+      command,
+      'local deleted = {}',
+      'local jobTS',
+      'for _, job in ipairs(JOBS) do',
+      '  if (redis.call("EXISTS", ARGV[1] .. job ..  ":lock") == 0) then',
```

Comment:
I see you are generating the key in LUA, this is actually not recommended by redis. Also, I don't know how fast is string concatenation in LUA, but it is better to make the concatenation once instead of for every loop iteration.